staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -614,11 +614,18 @@ func (r *crdHandler) getOrCreateServingInfoFor(uid types.UID, name string) (*crd
 		if val == nil {
 			continue
 		}
-		structuralSchemas[v.Name], err = structuralschema.NewStructural(val.OpenAPIV3Schema)
+		s, err := structuralschema.NewStructural(val.OpenAPIV3Schema)
 		if *crd.Spec.PreserveUnknownFields == false && err != nil {
-			utilruntime.HandleError(err)
+			// This should never happen. If it does, it is a programming error.
+			utilruntime.HandleError(fmt.Errorf("failed to convert schema to structural: %v", err))
+			return nil, fmt.Errorf("the server could not properly serve the CR schema") // validation should avoid this
+		}
+		if err := structuraldefaulting.PruneDefaults(s); err != nil {
```

Comment:
PruneDefaults can mutate subelements of the defaults in the passed-in schema... do we own the defaults object we're passing in? it looks like NewStructural makes a shallow copy, and the `crd` we have here is the one from the lister cache

Reply:
I don't think we deepcopy default. I better add an explicit deepcopy here. It will not harm, and makes the code less sensitive to later changes.

Reply:
done